File: test/tests/qualifiers.js
const { assert } = require('chai');

it('sync', () => {
  const o = new dll.Qualified;
  assert.strictEqual(o.get1(), 1);
  assert.strictEqual(o.get2(), 2);
  assert.strictEqual(o.get3(), 3);
  assert.strictEqual(dll.Qualified.get4(), 4);
  assert.strictEqual(dll.globalQualified(), 0);
});

it('async', (done) => {
  const o = new dll.Qualified;
  o.get1Async()
    .then((r) => assert.strictEqual(r, 1))
    .then(() => o.get2Async())
    .then((r) => assert.strictEqual(r, 2))
    .then(() => o.get3Async())
    .then((r) => assert.strictEqual(r, 3))
    .then(() => dll.Qualified.get4Async())
    .then((r) => assert.strictEqual(r, 4))
    .then(() => dll.globalQualified())
    .then((r) => assert.strictEqual(r, 0))
    .then(() => done())
    .catch(done);
});

File: test/tests/qualifiers.cc
#include <fixtures/qualifiers.h>

#include <nobind.h>

NOBIND_MODULE(qualifiers, m) {
  m.def<Qualified>("Qualified")
      .cons<>()
      .def<&Qualified::get1>("get1")
      .def<&Qualified::get2>("get2")
      .def<&Qualified::get3>("get3")
      .def<&Qualified::static_get4>("get4")
      .def<&Qualified::get1, Nobind::ReturnAsync>("get1Async")
      .def<&Qualified::get2, Nobind::ReturnAsync>("get2Async")
      .def<&Qualified::get3, Nobind::ReturnAsync>("get3Async")
      .def<&Qualified::static_get4, Nobind::ReturnAsync>("get4Async");

  m.def<&GlobalQualified>("globalQualified");
  m.def<&GlobalQualified, Nobind::ReturnAsync>("globalQualifiedAsync");
}
